crates/bevy_math/src/shape_sampling.rs

```diff
@@ -0,0 +1,238 @@
+                point + Vec2::Y * self.half_length
+            } else {
+                point - Vec2::Y * self.half_length
+            }
+        }
+    }
+
+    fn sample_surface<R: Rng + ?Sized>(&self, rng: &mut R) -> Vec2 {
+        let rectangle_surface = 4.0 * self.half_length;
+        let capsule_surface = rectangle_surface + TAU * self.radius;
+        if rng.gen_bool((rectangle_surface / capsule_surface) as f64) {
+            let side_distance = rng.gen_range((-2.0 * self.half_length)..=(2.0 * self.half_length));
+            if side_distance < 0.0 {
+                Vec2::new(self.radius, side_distance + self.half_length)
+            } else {
+                Vec2::new(-self.radius, side_distance - self.half_length)
+            }
+        } else {
+            let circle = Circle::new(self.radius);
+            let point = circle.sample_surface(rng);
+            // Add half length if it is the top semi-circle, otherwise subtract half
+            if point.y > 0.0 {
+                point + Vec2::Y * self.half_length
+            } else {
+                point - Vec2::Y * self.half_length
+            }
+        }
+    }
+}
+
+impl ShapeSample for Capsule3d {
+    type Output = Vec3;
+
+    fn sample_volume<R: Rng + ?Sized>(&self, rng: &mut R) -> Vec3 {
+        let cylinder_vol = PI * self.radius * self.radius * 2.0 * self.half_length;
+        // Add 4/3 pi r^3
+        let capsule_vol = cylinder_vol + 4.0 / 3.0 * PI * self.radius * self.radius * self.radius;
+        // Check if the random point should be inside the cylinder
+        if rng.gen_bool((cylinder_vol / capsule_vol) as f64) {
+            self.to_cylinder().sample_volume(rng)
+        } else {
+            let sphere = Sphere::new(self.radius);
+            let point = sphere.sample_volume(rng);
+            // Add half length if it is the top semi-sphere, otherwise subtract half
+            if point.y > 0.0 {
+                point + Vec3::Y * self.half_length
+            } else {
+                point - Vec3::Y * self.half_length
+            }
+        }
+    }
+
+    fn sample_surface<R: Rng + ?Sized>(&self, rng: &mut R) -> Vec3 {
+        let cylinder_surface = TAU * self.radius * 2.0 * self.half_length;
+        let capsule_surface = cylinder_surface + 4.0 * PI * self.radius * self.radius;
+        if rng.gen_bool((cylinder_surface / capsule_surface) as f64) {
+            let Vec2 { x, y: z } = Circle::new(self.radius).sample_surface(rng);
+            let y = rng.gen_range(-self.half_length..=self.half_length);
+            Vec3::new(x, y, z)
+        } else {
+            let sphere = Sphere::new(self.radius);
+            let point = sphere.sample_surface(rng);
+            // Add half length if it is the top semi-sphere, otherwise subtract half
+            if point.y > 0.0 {
+                point + Vec3::Y * self.half_length
+            } else {
+                point - Vec3::Y * self.half_length
+            }
+        }
+    }
+}
```

Comment:
Same thing as for `Cylinder`; maybe clarifying which way this is oriented would be good somewhere.